src/utils/index.ts

```diff
@@ -50,6 +50,24 @@ export function areSameAddress(a: string, b: string) {
   return a.toUpperCase() === b.toUpperCase();
 }
 
+export function isValidBytes32CompliantString(value: string, variableName: string) {
```

Comment:
These two functions should be combined into one:

- Call it `checkStringLength` (having the name start with `is` implies that the function returns a boolean, which it doesn't)
- Have it receive an optional `opts` object with `minLength: number | undefined` and `maxLength: number` properties (you can default them to 0 and 32 or whatever sensible default you deem appropriate)
- If `minLength` is 0 or undefined, it means it can be an empty string and the error message should say `... up to ${maxLength} characters long`
- If `minLength` is a positive number, it can't be an empty string and the error message should say `... between ${minLength} and ${maxLength} characters long`
- Write tests for it